servers/physics_2d/area_2d_sw.cpp

```diff
@@ -215,7 +215,9 @@ void Area2DSW::call_queries() {
 
 		for (Map<BodyKey, BodyState>::Element *E = monitored_bodies.front(); E;) {
 			if (E->get().state == 0) { // Nothing happened
-				E = E->next();
+				Map<BodyKey, BodyState>::Element *next = E->next();
```

Comment:
What about moving these new 3 lines in front of the if statement and then removing duplicated lines below? Since you are erasing them every iteration now anyway.

Reply:
I guess you'd have to change a bit more surrounding code then, so maybe not worth it.

Reply:
The problem is we can't assign `E` until we've erased it, and we can't erase `E` until we've extracted the information from it. We also don't want to extract the information from `E` if its `state` is 0. So at best we can move the assign `next` line before the `if` statement. However, I think it's clearer if the assigning `next` is kept close to assigning `E` to `next`, even if it means the line is duplicated later.